api/applications.go

```diff
@@ -37,15 +39,103 @@ type ApplicationListResponse struct {
 	Icon        string `json:"icon"`
 }
 
+// DeleteApplications delete application
+func DeleteApplications(c *gin.Context) {
+	idParam := c.Param("id")
+	id, err := strconv.ParseUint(idParam, 10, 32)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Invalid id=%q", idParam),
+			Error:   err.Error(),
+		})
+		return
+	}
+	// Build application model TODO refactor
+	db := model.GetDB()
+	app := &model.Application{
+		ID: uint(id),
+	}
+	var deployments []*model.Deployment
+	db.Model(app).Related(&deployments, "Deployments")
+	log.Debugf("Associated deployments: %#v", deployments)
+	organization := auth.GetCurrentOrganization(c.Request)
+	err = db.Model(organization).Related(app).Error
+	if err != nil {
+		log.Errorf("Error getting application: %s", err.Error())
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting application",
+			Error:   err.Error(),
+		})
+		return
+	}
+	app.Deployments = deployments
+
+	// Get cluster for application
+	filter := make(map[string]interface{})
+	filter["organization_id"] = organization
+	filter["id"] = app.ClusterID
+	commonCluster, ok := GetCommonClusterFromFilter(c, filter)
+	if !ok {
+		return
+	}
+	config, err := commonCluster.GetK8sConfig()
+	if err != nil {
+		log.Errorf("Error getting cluster config: %s", err.Error())
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting cluster config",
+			Error:   err.Error(),
+		})
+		return
+	}
+	application.DeleteApplication(app, config)
+}
+
+// ApplicationDetails get application details
+func ApplicationDetails(c *gin.Context) {
+	idParam := c.Param("id")
+	id, err := strconv.ParseUint(idParam, 10, 32)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Invalid id=%q", idParam),
+			Error:   err.Error(),
+		})
+		return
+	}
+	db := model.GetDB()
+	application := &model.Application{
+		ID: uint(id),
+	}
+	var deployments []*model.Deployment
```

Comment:
This section is similar to what's being used in other methods. Please factor out into separate function.